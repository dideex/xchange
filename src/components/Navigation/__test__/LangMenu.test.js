import React from 'react'
import {shallow, mount} from 'enzyme'
import {Provider as MobxProvider} from 'mobx-react'

import Component from '../LangMenu'
import UserStore from '../../../store/User'

describe('Navigation: lang menu', () => {
  let userStore = new UserStore()
  beforeEach(() => {
    userStore = new UserStore()
  })
  describe('Markup', () => {
    it('Basic markup', () => {
      const wrapper = shallow(
        <MobxProvider>
          <Component userStore={userStore} />
        </MobxProvider>,
      )

      expect(wrapper.html()).toMatchSnapshot()
    })
  })
  describe('Component behaviour', () => {
    it('Should change langauge to rus', () => {
      userStore.changeLocale = jest.fn()
      const wrapper = mount(
        <MobxProvider>
          <Component userStore={userStore} />
        </MobxProvider>,
      )
      wrapper.find('[data-testid="ru-RU"]').simulate('click')

      expect(userStore.changeLocale).toHaveBeenCalledTimes(1)
      expect(userStore.changeLocale).toHaveBeenCalledWith('ru-RU')
    })

    it('Should change langauge to eng', () => {
      userStore.changeLocale = jest.fn()
      const wrapper = mount(
        <MobxProvider>
          <Component userStore={userStore} />
        </MobxProvider>,
      )
      wrapper.find('[data-testid="en-US"]').simulate('click')

      expect(userStore.changeLocale).toHaveBeenCalledTimes(1)
      expect(userStore.changeLocale).toHaveBeenCalledWith('en-US')
    })

    it('Should change langauge to fr', () => {
      userStore.changeLocale = jest.fn()
      const wrapper = mount(
        <MobxProvider>
          <Component userStore={userStore} />
        </MobxProvider>,
      )
      wrapper.find('[data-testid="fr-FR"]').simulate('click')

      expect(userStore.changeLocale).toHaveBeenCalledTimes(1)
      expect(userStore.changeLocale).toHaveBeenCalledWith('fr-FR')
    })

    it('Should change langauge to cn', () => {
      userStore.changeLocale = jest.fn()
      const wrapper = mount(
        <MobxProvider>
          <Component userStore={userStore} />
        </MobxProvider>,
      )
      wrapper.find('[data-testid="zh-ZH"]').simulate('click')

      expect(userStore.changeLocale).toHaveBeenCalledTimes(1)
      expect(userStore.changeLocale).toHaveBeenCalledWith('zh-ZH')
    })

  })
})
