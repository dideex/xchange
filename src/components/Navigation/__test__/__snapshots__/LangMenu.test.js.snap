// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation: lang menu Markup Basic markup 1`] = `"<span data-testid=\\"ru-RU\\">Рус</span><span data-testid=\\"en-US\\">Eng</span><span data-testid=\\"fr-FR\\">Fr</span><span data-testid=\\"zh-ZH\\">Cn</span>"`;
